snippets/nrf91-modem-trace-ext-flash/snippet.yml

```diff
@@ -0,0 +1,11 @@
+name: nrf91-modem-trace-ext-flash
+append:
+  EXTRA_CONF_FILE: modem-trace-ext-flash.conf
+  EXTRA_DTC_OVERLAY_FILE: modem-trace-uart.overlay
+boards:
+  nrf9160dk_nrf9160_ns:
+    append:
+      EXTRA_DTC_OVERLAY_FILE: nrf9160dk-ext-flash.overlay
+  /nrf91[356]1[de]k_nrf91[356]1_ns/:
```

Comment:
Yes, it saves us a bit of duplication, but I'm not sure if it's a good idea or not. It could be useful to extract the boards we support for each snippet to list in the docs, like we're doing from `sample.yaml`, and the regex would make that harder. However, I think it's fine for now, and it's easy to change later if we regret it.